ldap_auth_provider.py

```diff
@@ -84,6 +86,9 @@ def __init__(self, config, account_handler):
         self.ldap_active_directory = config.active_directory
         if self.ldap_active_directory:
             self.ldap_default_domain = config.default_domain
+            # This will be set to value of Active Directory root domain of type str
+            # or to None in case of error while retrieving Active Directory root domain
+            self.ldap_root_domain = False
```

Comment:
you could also give this a type annotation. Currently it would be:

```python
            self.ldap_root_domain = False   # type: Union[Literal[False],str,None]
```

though as above I think `Optional[str]` would be easier to understand.